kolibri/plugins/device/assets/src/views/ManageContentPage/TasksBar.vue

```diff
@@ -81,6 +75,14 @@
         return this.$tr('someTasksComplete', { done: this.doneTasks, total: this.totalTasks });
       },
     },
+    watch: {
+      doneTasks(val, oldVal) {
+        // Just refresh the channel list whenever anything finishes to get the latest version
+        if (val > oldVal) {
+          this.$store.dispatch('manageContent/refreshChannelList');
+        }
+      },
+    },
```

Comment:
It seems a bit odd to put this logic inside a watcher in a vue component. I would have expected it to live inside  the handler from the `refreshTaskList` action:

https://github.com/learningequality/kolibri/blob/aecc35cd267ea15d8e4e4bf30ecdcfcd90c29d92/kolibri/plugins/device/assets/src/modules/manageContent/actions/taskActions.js#L37-L47

Some of the computed props in this component might be implemented as vuex getters instead to help abstract away the task implementation details stuff.

related: https://github.com/learningequality/kolibri/issues/6177

Reply:
I'd rather not put the refresh channel side effect in refreshTaskList, since we would potentially be updating the channel list in places other than the ManageContentPage. We'd need to think about how to handle sudden changes to the channel list like in Delete/Export flows.

Reply:
In practice, I've found that having the channel list in vuex and globally shared is not always ideal and that each component should request and manage the channel list themselves for their own purposes. Under this structure, the ManageContentPage would still be watching the Task queue and refreshing its local copy of the channels.

Reply:
sounds good